tests/mail_test.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+
+include './src/mail.sh'
+include './tests/utils.sh'
+
+function oneTimeSetUp()
+{
+  declare -gr ORIGINAL_DIR="$PWD"
+  declare -gr FAKE_GIT="$SHUNIT_TMPDIR/fake_git/"
+
+  mk_fake_git "$FAKE_GIT"
+}
+
+function oneTimeTearDown()
+{
+  rm -rf "$FAKE_GIT"
+}
+
+function setUp()
+{
+  declare -gA options_values
+
+  cd "$SHUNIT_TMPDIR" || {
+    ret="$?"
+    fail "($LINENO): Failed to move to shunit temp dir"
+    exit "$ret"
+  }
+}
+
+function tearDown()
+{
+  unset options_values
+}
+
+function test_mail_parser()
+{
+  local output
+  local expected_result
+  local ret
+
+  # Invalid options
+  parse_mail_options '--smtpencryption' 'tlst' &> /dev/null
+  expected_result=''
+  assert_equals_helper 'Encryption should be blank' "$LINENO" "${options_values['SMTPENCRYPTION']}" "$expected_result"
+
+  parse_mail_options '--wrongOption' 'lala xpto'
+  ret="$?"
+  assert_equals_helper 'Invalid option passed' "$LINENO" "$ret" 22
+
+  # valid options
+  parse_mail_options '--setup'
+  expected_result=1
+  assert_equals_helper 'Set setup flag' "$LINENO" "${options_values['SETUP']}" "$expected_result"
+
+  parse_mail_options '--global'
+  expected_result='--global'
+  assert_equals_helper 'Set global flag' "$LINENO" "${options_values['GLOBAL']}" "$expected_result"
+
+  parse_mail_options '--name' 'Xpto Lala'
+  expected_result="user.name 'Xpto Lala'"
+  assert_equals_helper 'Set name' "$LINENO" "${options_values['NAME']}" "$expected_result"
+
+  parse_mail_options '--email' 'test@email.com'
+  expected_result="user.email 'test@email.com'"
+  assert_equals_helper 'Set email' "$LINENO" "${options_values['EMAIL']}" "$expected_result"
+
+  parse_mail_options '--smtpuser' 'test@email.com'
+  expected_result="sendemail.smtpuser 'test@email.com'"
+  assert_equals_helper 'Set smtp user' "$LINENO" "${options_values['SMTPUSER']}" "$expected_result"
+
+  parse_mail_options '--smtpencryption' 'tls'
+  expected_result="sendemail.smtpencryption 'tls'"
+  assert_equals_helper 'Set smtp encryption to tls' "$LINENO" "${options_values['SMTPENCRYPTION']}" "$expected_result"
+
+  parse_mail_options '--smtpencryption' 'ssl'
+  expected_result="sendemail.smtpencryption 'ssl'"
+  assert_equals_helper 'Set smtp encryption to ssl' "$LINENO" "${options_values['SMTPENCRYPTION']}" "$expected_result"
+
+  parse_mail_options '--smtpserver' 'test.email.com'
+  expected_result="sendemail.smtpserver 'test.email.com'"
+  assert_equals_helper 'Set smtp server' "$LINENO" "${options_values['SMTPSERVER']}" "$expected_result"
+
+  parse_mail_options '--smtpserverport' '123'
+  expected_result="sendemail.smtpserverport '123'"
+  assert_equals_helper 'Set smtp serverport' "$LINENO" "${options_values['SMTPSERVERPORT']}" "$expected_result"
+
+  parse_mail_options '--smtppass' 'verySafePass'
+  expected_result="sendemail.smtppass 'verySafePass'"
+  assert_equals_helper 'Set smtp pass' "$LINENO" "${options_values['SMTPPASS']}" "$expected_result"
```

Comment:
What happens if we try to use `--smtpxpto`? I suppose we will accept it, right?

Reply:
Parser fails, added a test just in case.